flask/app/variants.py

```diff
@@ -227,17 +226,109 @@ def parse_gene_panel() -> List[Any]:
         app.logger.error("Not all requested genes were found.")
         abort(400, description="Not all requested genes were found.")
 
-    genes = query.all()
+    genes_list = query.all()  # TODO: is this line necessary?
+
+    return gene_filter
+
+
+def parse_region(region: str):
+    """
+    Parses a given list of ranges (eg. "chr1:0500-0509,chr5:0600-0630,chr18:0440-0330").
+    Returns a filter for use against the variant table to find corresponding variants
+    based on gene start and end positions.
+    """
+    CHR, START, END = 0, 1, 2
+    # format check
+    # 1: chromosome, 2: start position, 3: end position
+    region_pattern = re.compile("chr([\da-zA-Z]+):([\d]+)-([\d]+)")
+    matches = [region_pattern.match(r) for r in region.split(",")]
+
```

Comment:
This isn't necessary, but I think input validation such that the requested end position is > start position would be nice.

Reply:
The parsing function is already agnostic to the start-end order. Lines 253-257 correct the order if the start position is higher than the end position.

Reply:
Ah I see, in that case then eg. `http://localhost:5000/api/summary/participants?panel=chr10:100028007-100007446&search_by=region` and `http://localhost:5000/api/summary/participants?panel=chr10:100007446-100028007&search_by=region` should return the same results? If so I get 0 results back on the former

Reply:
It looks like the set itself isn't being modified with the corrected order, so the set that's iterated over to generate the filters isn't actually corrected. A solution might be to convert the set to a list and index into the list for assignment eg. `for i, r in enumerate(region_list)`